.pre-commit-config.yaml

```diff
@@ -7,6 +7,7 @@ repos:
   rev: v0.0.281
   hooks:
   - id: ruff
+    exclude: gepetuto/magic.py
```

Comment:
This won't prevent manual runs of `ruff .` to fail. I think you should instead add hints in the file : either a `  # noqa: F821` in the end of each line where there is a `get_ipython`, or a `# ruff: noqa: F821` at the begining of the file.

A better solution might be to add a `def run(self, magic)` method, which run `get_ipython().run_line_magic(magic)` with the correct noqa only once, and use this method in the other places.